File: apple/Sources/ReactNativeWatchOS/Modifiers/SimultaneousGesture.swift
import SwiftUI

/// SwiftUI `.simultaneousGesture(_:)`. Recognizes alongside the view's own
/// gestures. Same v1 limitation as `gesture`: only `tap`/`longPress` bridge.
enum RNWSimultaneousGestureModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("simultaneousGesture") { view, params, ctx in
            RNWGestureSupport.apply(.simultaneous, to: view, params: params, ctx: ctx)
        }
    }
}

File: src/modifiers/simultaneousGesture.ts
import { createModifier } from './createModifier';
import type { GestureParams } from './gesture';

/// SwiftUI `.simultaneousGesture(_:)`. Attaches a gesture that recognizes
/// alongside the view's own (and any ancestor/descendant) gestures rather
/// than competing with them.
///
/// **Limited (v1):** same bridging constraints as `gesture` — only `'tap'`
/// and `'longPress'` are supported; arbitrary SwiftUI `Gesture` values are
/// not expressible across the bridge.
export function simultaneousGesture(params: GestureParams) {
  return createModifier('simultaneousGesture', params);
}

export type { GestureParams, GestureType } from './gesture';
